host/volume/api/http.go

```diff
@@ -0,0 +1,114 @@
+package volumeapi
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/julienschmidt/httprouter"
+	"github.com/flynn/flynn/host/volume"
+	"github.com/flynn/flynn/host/volume/zfs"
+	"github.com/flynn/flynn/pkg/httphelper"
+	"github.com/flynn/flynn/pkg/random"
+)
+
+type HttpAPI struct {
+	vman *volume.Manager
+}
+
+func NewHttpAPI(vman *volume.Manager) *HttpAPI {
+	return &HttpAPI{vman: vman}
+}
+
+func RegisterRoutes(api *HttpAPI, r *httprouter.Router) {
+	r.POST("/volume/provider", api.CreateProvider)
+	r.POST("/volume/provider/:provider_id/newVolume", api.Create)
+	r.PUT("/volume/instances/:id/snapshot", api.Snapshot)
+	r.GET("/volume/instances/:id/inspect", api.Inspect)
```

Comment:
I think this API should be something along these lines (not totally sold on `volume_providers`)

```text
GET  /volume_providers - list volume providers
GET  /volume_providers/:provider_id/volumes - list volumes
POST /volume_providers/:provider_id/volumes - create volume
GET  /volume_providers/:provider_id/volumes/:volume_id - get volume details
POST /volume_providers/:provider_id/volumes/:volume_id/snapshots - create snapshot
```